core/src/main/java/com/linecorp/armeria/server/docs/DocStringExtractor.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.reflections.Configuration;
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+import org.reflections.util.FilterBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Resources;
+
+/**
+ * A supporting base class for implementing the standard pattern of extracting docstrings
+ * from arbitrary files in a particular classpath location.
+ */
+public abstract class DocStringExtractor {
+
+    private static final Logger logger = LoggerFactory.getLogger(DocStringExtractor.class);
+
+    private static final Map<ClassLoader, Map<String, String>> cached = new ConcurrentHashMap<>();
```

Comment:
Sounds like a good idea, how about I come back to this after we add Caffeine, which will make it trivial?